src/uniprotkb/components/landing-page/StatisticsChart.tsx

```diff
@@ -0,0 +1,241 @@
+import * as d3 from 'd3';
+import { useHistory } from 'react-router-dom';
+
+import useDataApi from '../../../shared/hooks/useDataApi';
+
+import {
+  StatisticsItem,
+  StatisticsPayload,
+} from '../statistics-page/StatisticsPage';
+import { LocationToPath, Location } from '../../../app/config/urls';
+
+const renderPieChart = (
+  svgElement: SVGSVGElement | null,
+  data: StatisticsItem[],
+  history
+): void => {
+  // Specify the chart’s dimensions.
+  const width = 400,
+    height = 300,
+    margin = 40;
+
+  const radius = Math.min(width, height) / 2 - margin;
+
+  // Create the color scale.
+  const color = d3
+    .scaleOrdinal()
+    .domain(data.map((d) => d.name))
+    .range([...d3.schemeBlues[5]].reverse());
+
+  // Create the SVG container.
+  const svg = d3
+    .select(svgElement)
+    .attr('width', width)
+    .attr('height', height)
+    .append('g')
+    .attr('transform', 'translate(' + width / 2 + ',' + height / 2 + ')');
+
+  const g = svg.append('g');
+
+  g.append('g').attr('class', 'slices');
+  g.append('g').attr('class', 'labels');
+  g.append('g').attr('class', 'lines');
+
+  // Create the pie layout and arc generator.
+  const pie = d3
+    .pie<StatisticsItem>()
+    .sort(null)
+    .value((d) => d.count);
+
+  const arc = d3
+    .arc()
+    .outerRadius(radius * 0.8)
+    .innerRadius(0);
+
+  const outerArc = d3
+    .arc()
+    .innerRadius(radius * 0.9)
+    .outerRadius(radius * 0.9);
+
+  const key = (d: d3.PieArcDatum<StatisticsItem>) => {
+    return (d.data as StatisticsItem).name;
+  };
+
+  /* ------- PIE SLICES -------*/
+  const pieData = pie(data);
+  const slice = g
+    .select('.slices')
+    .selectAll<SVGPathElement, d3.PieArcDatum<StatisticsItem>>('path.slice')
+    .data(pieData, key);
+
+  slice
+    .enter()
+    .insert('path')
+    .attr('class', 'slice')
+    .merge(slice)
+    .transition()
+    .duration(1000)
+    .attrTween('d', function (d) {
+      this._current = this._current || d;
+      const interpolate = d3.interpolate(this._current, d);
+      this._current = interpolate(0);
+      return function (t) {
+        return arc(interpolate(t));
+      };
+    })
+    .style('fill', (d) => {
+      return color(d.data.name);
+    })
+    .style('stroke-width', '2px');
+
+  slice.exit().remove();
+
+  /* ------- TEXT LABELS -------*/
+
+  const text = g
+    .select('.labels')
+    .selectAll<SVGTextElement, d3.PieArcDatum<StatisticsItem>>('text')
+    .data(pie(data));
+
+  const midAngle = (d: d3.DefaultArcObject) => {
+    return d.startAngle + (d.endAngle - d.startAngle) / 2;
+  };
+
+  text
+    .enter()
+    .append('text')
+    .attr('dy', '.35em')
+    .style('pointer-events', 'auto')
+    .style('cursor', 'pointer')
+    .style('fill', '#014371')
+    .style('font-weight', 'bold')
+    .on('click', function (d) {
+      history.push({
+        pathname: LocationToPath[Location.UniProtKBResults],
+        search: `query=(taxonomy_name:${d.data.name})`,
```

Comment:
Might be good to use Dan's new URLSearchParameters wrapper that would automatically escape whatever needs escaping (I think either the parenthesis or the colon, or both)